### PR TITLE
shipit-static-analysis: Clang format

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: src/shipit_static_analysis/tests/test_clang.py [hunk 1]
```diff
@@ -1,13 +1,20 @@
 # -*- coding: utf-8 -*-
-from shipit_static_analysis.clang import ClangIssue
+from shipit_static_analysis.clang.tidy import ClangTidyIssue
+from shipit_static_analysis.clang.format import ClangFormat, ClangFormatIssue
+
+BAD_CPP_OLD = '#include <demo>\nint \tmain(void){\n printf("plop");return 42;    '
+BAD_CPP_NEW = '''#include <demo>
+int main(void) {
+  printf("plop");
+  return 42;'''
```
Comment (jankeromnes) [inline src/shipit_static_analysis/tests/test_clang.py]: Not: Please use the same string formatting in both constants to facilitate visual comparison.

File: src/shipit_static_analysis/tests/test_clang.py [hunk 2]
```diff
@@ -16,14 +23,43 @@ def test_expanded_macros():
...
+
+
+def test_clang_format(tmpdir):
+    '''
+    Test clang-format runner
+    '''
+
+    # Write badly formatted c file
+    bad_file = tmpdir.join('bad.cpp')
+    bad_file.write('''#include <demo>\nint \tmain(void){\n printf("plop");return 42;    \n}''')
```
Comment (jankeromnes) [inline src/shipit_static_analysis/tests/test_clang.py]: Nit: Why not use the above constant here?

File: src/shipit_static_analysis/tests/test_clang.py (hunk 1 above)
Comment (jankeromnes) [reply src/shipit_static_analysis/tests/test_clang.py]: Not "Not" but "Nit". Thank you, auto-correct.

Comment (La0) [reply src/shipit_static_analysis/tests/test_clang.py]: :+1:

File: src/shipit_static_analysis/shipit_static_analysis/clang/format.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        return self.in_patch
+
+    @property
+    def mozreview_body(self):
+        '''
+        Build the text body published on MozReview
+        According to diff mode
+        '''
+        if self.mode == OPCODE_REPLACE:
+            return 'Replace by: \n\n```\n{}\n```'.format(self.new)
```
[the remark this answers is not in the document]
Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/format.py]: :+1:

File: src/shipit_static_analysis/shipit_static_analysis/workflow.py [hunk 4]
```diff
@@ -5,31 +5,48 @@
...
 C/C++ static analysis didn't find any defects in this patch. Hooray!
 '''
 MOZREVIEW_COMMENT_FAILURE = '''
-C/C++ static analysis found {} defect{} in this patch{}.
+C/C++ static analysis found {total} defect{total_s} in this patch{extras_comments}.
+ - {nb_clang_tidy} defects found by clang-tidy
+ - {nb_clang_format} defects found by clang-format
 
-You can run this analysis locally with: `./mach static-analysis check path/to/file.cpp`
+You can run this analysis locally with: `./mach static-analysis check path/to/file.cpp` and `./mach clang-format path/to/file.cpp`
```
Comment (sylvestre) [inline src/shipit_static_analysis/shipit_static_analysis/workflow.py]: clang-format -p path/

File: src/shipit_static_analysis/shipit_static_analysis/workflow.py [hunk 5]
```diff
@@ -5,31 +5,48 @@
...
+ARTIFACT_URL = 'https://queue.taskcluster.net/v1/task/{task_id}/runs/{run_id}/artifacts/public/results/{diff_name}'
 MAX_COMMENTS = 30
 MOZREVIEW_COMMENT_SUCCESS = '''
 C/C++ static analysis didn't find any defects in this patch. Hooray!
 '''
 MOZREVIEW_COMMENT_FAILURE = '''
-C/C++ static analysis found {} defect{} in this patch{}.
+C/C++ static analysis found {total} defect{total_s} in this patch{extras_comments}.
+ - {nb_clang_tidy} defects found by clang-tidy
+ - {nb_clang_format} defects found by clang-format
```
Comment (sylvestre) [inline src/shipit_static_analysis/shipit_static_analysis/workflow.py]: "defects" => nits

**Pull request merged**